tests/components/onkyo/test_config_flow.py

```diff
@@ -222,6 +223,122 @@ async def test_discovery_with_one_selected(hass: HomeAssistant) -> None:
         assert select_result["description_placeholders"]["name"] == "type 42 (host 42)"
 
 
+async def test_ssdp_discovery_success(hass: HomeAssistant) -> None:
+    """Test SSDP discovery with valid host."""
+    discovery_info = ssdp.SsdpServiceInfo(
+        ssdp_location="http://192.168.1.100:8080",
+        upnp={ssdp.ATTR_UPNP_FRIENDLY_NAME: "Onkyo Receiver"},
+        ssdp_usn="uuid:mock_usn",
+        ssdp_udn="uuid:00000000-0000-0000-0000-000000000000",
+        ssdp_st="mock_st",
+    )
+
+    with patch(
+        "homeassistant.components.onkyo.config_flow.async_interview",
```

Comment:
Please patch pyeiscp instead of the Home Assistant code

Reply:
We decided to patch `async_interview` everywhere in tests, as it is a much cleaner interface. In the long run we intend to move it to the library.

Reply:
Let me know, happy to do either. But since all other tests in the file do this, I would lean towards making a separate PR and updating all the tests at once, not just this test.

Reply:
Feel free to ping me in the other PR as otherwise I won't see it

Reply:
@bdraco 
i've raised a PR to refactor the tests to patch the underlying library instead of HA here:
https://github.com/home-assistant/core/pull/132653

Reply:
Rebased and updated tests